cedar-policy-validator/src/diagnostics/validation_errors.rs

```diff
@@ -448,6 +449,114 @@ impl Diagnostic for HierarchyNotRespected {
     }
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq, Error, Copy)]
+/// Represents how many entity dereferences can be applied to a node.
+/// `None` represents infinity.
+pub struct EntityDerefLevel {
+    /// `None` represents infinity.
+    pub level: Option<i64>,
```

Comment:
Added a comment explaining what negative values represent.

`None` is no longer constructed, but I'd like to keep the code this way in case we add `level` back into the schema.

Reply:
I'd rather get rid of it and add it later if we need to, but I don't feel too strongly about it

Reply:
Done